process/economics/economicsData.go

```diff
@@ -674,21 +702,28 @@ func isTooMuchGasProvided(gasProvided uint64, gasRemained uint64) bool {
 
 // ComputeTxFeeBasedOnGasUsed will compute transaction fee
 func (ed *economicsData) ComputeTxFeeBasedOnGasUsed(tx data.TransactionWithFeeHandler, gasUsed uint64) *big.Int {
+	currenEpoch := ed.enableEpochsHandler.GetCurrentEpoch()
+	return ed.ComputeTxFeeBasedOnGasUsedInEpoch(tx, gasUsed, currenEpoch)
+}
+
+// ComputeTxFeeBasedOnGasUsedInEpoch will compute transaction fee in a specific epoch
+func (ed *economicsData) ComputeTxFeeBasedOnGasUsedInEpoch(tx data.TransactionWithFeeHandler, gasUsed uint64, epoch uint32) *big.Int {
 	moveBalanceGasLimit := ed.ComputeGasLimit(tx)
 	moveBalanceFee := ed.ComputeMoveBalanceFee(tx)
 	if gasUsed <= moveBalanceGasLimit {
 		return moveBalanceFee
 	}
 
-	computeFeeForProcessing := ed.ComputeFeeForProcessing(tx, gasUsed-moveBalanceGasLimit)
+	computeFeeForProcessing := ed.ComputeFeeForProcessingInEpoch(tx, gasUsed-moveBalanceGasLimit, epoch)
 	txFee := big.NewInt(0).Add(moveBalanceFee, computeFeeForProcessing)
 
 	return txFee
 }
 
 // EpochConfirmed is called whenever a new epoch is confirmed
 func (ed *economicsData) EpochConfirmed(epoch uint32, _ uint64) {
-	ed.statusHandler.SetStringValue(common.MetricGasPriceModifier, fmt.Sprintf("%g", ed.GasPriceModifier()))
+	ed.statusHandler.SetStringValue(common.MetricGasPriceModifier, fmt.Sprintf("%g", ed.GasPriceModifierInEpoch(epoch)))
```

Comment:
I think the new model needs to do something regarding the rewards config and gas limit config as well.

With the methods accepting an epoch, the rewards config and gas limit config will not be synced with the given epoch but with the confirmed epoch.

Reply:
updated as suggested